datashape/typesets.py

```diff
@@ -90,8 +89,9 @@ def __init__(self):
         self.lookup = self.registry.get
 
     def register_typeset(self, name, typeset):
-        if name in typeset:
-            raise DataShapeTypeError("TypeSet %s already defined" % name)
+        if name in self.registry:
```

Comment:
this looks more correct; however, I just want to make sure that the behaviour change is okay here.